controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -2213,11 +2215,11 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
                 switch (txnRecord.getStatus()) {
                     case COMMITTING:
                         if (txnRecord.getCommitOrder() == order) {
-                            // if entry matches record's position then include it
-                            transactionsMap.add(txnRecord);
-                            if (transactionsMap.size() >= limit) {
-                                break;
-                            }
+                             if (transactionsMap.size() < limit) {
+                                 // if entry matches record's position then include it
+                                 transactionsMap.add(txnRecord);
+                             }
+                             committingTxnCount.addAndGet(1);
```

Comment:
this is counting txns only in the lowest epoch. 
so either we fetch all txns which are found in orderer and check which of those are already marked for committing, which can be optimized by doing these fetches in parallel (present logic is trying to fetch in batches of upto `limit` as it is trying to create a batch of `limit` size only. but you are doing a count of all such txns so doing gets in parallel may be faster.

Reply:
Although i would suggest we dont do the exact count anyway (which can change while we are counting).
instead we can go with the approximate value which is simply zkorderer.children_count

in happy case if a txn is in orderer, then will also be marked as committing. so we dont need the accurate count but an approximate count that can tell whether there is a growing backlog or not.

Reply:
done